src/CoreWCF.Primitives/tests/ServiceKnownTypeTests.cs

```diff
@@ -0,0 +1,187 @@
+            TestHelper.CloseServiceModelObjects((System.ServiceModel.Channels.IChannel)channel, factory);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest_Ex()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingService_Ex, PingService_Ex>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            Assert.Throws<System.Xml.XmlException>(() =>
+            {
+                BaseMsg msg = channel.Ping("hello");
+            });
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest_Inline()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingService_Inline, PingService_Inline>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            BaseMsg msg = channel.Ping("hello");
+            Assert.Equal("hello", msg.msg);
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest_Inline_SSM()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingService_Inline_SSM, PingService_Inline_SSM>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            BaseMsg msg = channel.Ping("hello");
+            Assert.Equal("hello", msg.msg);
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest_Method()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingService_Method, PingService_Method>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            BaseMsg msg = channel.Ping("hello");
+            Assert.Equal("hello", msg.msg);
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest_Method_SSM()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingService_Method_SSM, PingService_Method_SSM>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            BaseMsg msg = channel.Ping("hello");
+            Assert.Equal("hello", msg.msg);
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+    }
+
+    internal class PingService_Ex : PingServiceBase, IPingService_Ex { }
+
+    [System.ServiceModel.ServiceContract]
+    public interface IPingService_Ex
+    {
+       
+        [System.ServiceModel.OperationContract]
+        BaseMsg Ping(string msg);
+
+    }
+
+    internal class PingService_Inline : PingServiceBase, IPingService_Inline { }
+   
+    [System.ServiceModel.ServiceContract]
+    [ServiceKnownType(typeof(DerivedMsg))]
+    [System.ServiceModel.ServiceKnownType(typeof(DerivedMsg))]
+    public interface IPingService_Inline
+    {
+       
+        [System.ServiceModel.OperationContract]
```

Comment:
There's still some issues with extra blank lines in the interfaces and in PingServiceBase. Once those are fixed, this looks good to merge.

Reply:
Done as per above